### PR TITLE
Move module-info directly into the build

### PR DESCRIPTION
**Pull request by hazendaz**
This is part of the puzzle.  This puts modular-info.java / modular-info.test directly into the build.  If a runtime using it is java 8, it doesn't get used.  If the runtime is java 11+, it does.

### CODE REVIEW

**Comment by hazendaz** [comment]
@dbwiddis This in combination with 'https://github.com/moditect/moditect' is likely what is needed to pull this off without having to create extra modules and to completely do the entire job.  I think the upstream on JNA probably needs the same.  This change by itself would not matter what java version is used to release.  It will only pick up the modular information in the build if java 11 or higher.

The only thing I see with the mvn11 (java11 profile) is trying to set binary at java 11.  But in reality there are so few libraries that have forced that and it doesn't really matter.  If you want binary at java 11, sure fine, but what about 100s of other libraries in a potential project that don't.  Its never mattered.  I wouldn't doubt some are java 4 compiled still and all still ok.

What I do know is the IDEs such as Eclipse that I use daily already throws everything in modular path as it is, regardless if the jar was or not.  It likely does this by adding automatic module naming on the fly when using higher jdks.

**Comment by hazendaz** [comment]
Signing off for night, but think this along with dropping java11 profile entirely probably would in fact give you all you were trying to do.  It would not pick up 'jpms' modules from jna but I feel like that ist just another case of compiled with java 11 targetting jdk 11 binary that doesn't quite mean much unless physically recoded to new java constructs or has something that is otherwise missing.  I could be 100% off base here but have also never really seen any direct modular usage in my day job.  No one seems to know how to do it nor seems to care and most still prefer java 8.

**Comment by dbwiddis** [comment]
This is the MRJAR approach. I considered this very early on -- it was really the simplest approach -- but after reading about some of the drawbacks, decided against it.  

The biggest concern for me is similar to our discussion on the other issue -- I don't think the JDK should automatically enforce new behavior.  Just because I'm running on JDK17 doesn't mean OSHI should force me to use modules.  Unfortunately I think with the approach in this PR, it will force me to use modules even if I don't want to, just because I've got a newer JDK.  So if I'm directly using one of OSHI's driver classes to access information not exposed in the official API, I'll get different behavior.

Yeah, I already hear the objection about implementation details/API/etc. that modules are supposed to enforce.  I'm reluctant to "enforce" that because I really don't see a good reason to do so.

I primarily want OSHI to remain primarily JDK8 non-module compatible.  I still don't think there's much of a use case for modules and am primarily providing the added artifact solely to support those who want to use those features, jlink, etc.

Here's an article from the Gradle team that helped sway me against MRJARs early on.  Interested in your thoughts on it: https://blog.gradle.org/mrjars

I think eventually ... another year or two... this may be the right approach, but I honestly still think Java hasn't figured all this stuff out. I've seen MRJAR issues as recently as JDK16.

File: mvnw11 [hunk 1]
```diff
@@ -32,8 +28,6 @@ echo "> ./mvnw \"$@\" -P java11"
 ./mvnw "$@" -P java11
```
Comment (dbwiddis) [inline mvnw11:28]: For the sake of review, I'll pick on something you didn't even change.  Test failures in this call aren't propagated to the script result, so the `mvn test` can fail gloriously but this script will happily exit with no problems.

File: oshi-core/src/test/resources/META-INF/versions/11/module-info.test [hunk 2]
```diff
@@ -22,4 +22,4 @@
 --add-reads
   com.github.oshi=org.hamcrest
 --add-reads
-  com.github.oshi=org.junit.jupiter.api
\ No newline at end of file
+  com.github.oshi=org.junit.jupiter.api
```
Comment (dbwiddis) [inline oshi-core/src/test/resources/META-INF/versions/11/module-info.test:25]: I'm curious why the newline at end of file is a requirement.

**Review by dbwiddis — COMMENTED** [review]
You asked for a review so I'm giving a review.   `mvnw11` is broken and should be totally eliminated.   MRJAR has some use cases but I don't think "adding a module descriptor" is one that we should leverage.

**Comment by dbwiddis** [comment]
With a full night's sleep and full dose of caffeine, I'll try to rewrite what I said above but more briefly:
1. A MRJAR is fine for users with java 8 only, or users with java 9+ who are only using API functions.
2. A MRJAR as proposed here would force any user with JDK9+ into having oshi on the module path along with the API restrictions that come with modules
3. There may be some users who use util classes/driver functions.  See #535 and #1238.  While those are non-API they are useful. 
4. I'm fine with the idea of an MRJAR for all variants of an already fully modular setup so newer versions can take advantage of newer features under the hood based on the user's runtime.   But during this transition period while supporting Java 8 I'm a pretty hard no on MRJAR.

**Pull request closed without merging**